test/serializers/links_test.rb

```diff
@@ -0,0 +1,62 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class LinksTest < Minitest::Test
+      def setup
+        ActionController::Base.cache_store.clear
+        @blog = Blog.new(id: 1,
+                         name: 'AMS Hints',
+                         writer: Author.new(id: 2, name: "Steve"),
+                         articles: [Post.new(id: 3, title: "AMS")])
+      end
+
+      def test_links_is_present_with_root
+        serializer = AlternateBlogSerializer.new(@blog, :links => {:self => "/blogs/1"})
+        adapter = ActiveModel::Serializer::Adapter::JsonApi.new(serializer)
```

Comment:
For future-proofing and consistency, we may want to instead do it the way the controller does (IIRC)

```ruby
serializable_resource = ActiveModel::SerializableResource.serialize(@blog, links: { self: "/blogs/1" }, adapter: :json_api, each_serializer: AlternateBlogSerializer)
serializable_resource.as_json
```